packages/vite/src/node/constants.ts

```diff
@@ -6,6 +6,21 @@ export const DEFAULT_MAIN_FIELDS = [
   'jsnext'
 ]
 
+/**
+ * A non-exhaustive list of known-to-be-ES-module entry names.
+ * From <https://github.com/stereobooster/package.json#package-bundlers>
+ */
+export const KNOWN_ESM_MAIN_FIELDS = [
+  'module',
+  'jsnext:main',
+  'jsnext',
+  'esnext',
+  'es2015',
+  'es2020',
+  'fesm2015',
+  'fesm2020'
```

Comment:
I didn't add `esm` and `module-browser` because I can't find any real-world use cases of them.
`esnext` is used by [`foundation-sites`](https://github.com/foundation/foundation-sites/blob/8846fdad0289d1a0a80929c92a4245fcb781c662/package.json#L7), `es20**` and `fesm20**` are used by Angular CLI.